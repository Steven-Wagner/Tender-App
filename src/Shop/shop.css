.shopping-page-info {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.done-button {
    padding: 5px;
    margin: 5px;
    font-size: 1.2em;
}

.shopping-page-info div {
    margin: 5px 5px;
}

.current-product {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.current-product h2 {
    margin: 0;
}

.product-pic {
    height: 25vh;
    max-width: 75%;
    border: 1px solid black;
    border-radius: 5px;
    box-shadow: 3px 3px 5px;
}

.current-product button {
    padding: 10px;
    margin: 10px;
    width: 40%;
    font-size: 1.2em;
}