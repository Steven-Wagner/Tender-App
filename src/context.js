import React from 'react';

const TenderContext = React.createContext();

export const ProductsProvider = TenderContext.Provider
export const ProductsConsumer = TenderContext.Consumer

export default TenderContext;