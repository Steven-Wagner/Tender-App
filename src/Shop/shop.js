import React, {Component} from 'react';
import TenderContext from '../context';
import CurrentProduct from './CurrentProduct/currentProduct';
import ShopInfoBar from './ShopInfoBar/shopInfoBar';

import './shop.css';

class Shop extends Component {
    
    static contextType = TenderContext;

    constructor(props) {
        super(props);

        this.handleBuy = this.handleBuy.bind(this);
        this.handleSkip = this.handleSkip.bind(this);
        this.handleLike = this.handleLike.bind(this);
        this.handleDone = this.handleDone.bind(this);
    }

    async handleBuy() {
        const canAfford = this.context.canAfford(this.context.currentShoppingItem.price)

        if (canAfford) {
            if (!this.context.currentShoppingItem.ad) {
                await this.context.removeItemFromState(this.context.currentShoppingItem.index);
            }
            await this.context.subtractTotalByPrice(this.context.currentShoppingItem.price);
            await this.context.newPurchasedItem(this.context.currentShoppingItem);
            await this.context.getNewProductToSell(0);
        }
        else {
            this.context.setPopupMessages('errorPopup', ['You can\'t afford that item'])
        }
    }

    handleSkip() {
        this.context.getNewProductToSell();
    }

    handleLike() {
        this.context.setCurrentProductToLiked();
    }

    handleDone() {
        this.context.goBack();
    }

    render() {
        return(
            <div>
                <ShopInfoBar 
                    totalMoney={this.context.totalMoney}
                    handleDone={this.handleDone}/>
                
                <CurrentProduct 
                    item={this.context.currentShoppingItem}
                    handleBuy={this.handleBuy}
                    handleSkip={this.handleSkip}
                    handleLike={this.handleLike}/>
            </div>
        )
    }
}

export default Shop;