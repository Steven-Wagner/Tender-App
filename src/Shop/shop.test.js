import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Shop from './shop';
import Enzyme, { mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {ProductsProvider} from '../context';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faShoppingCart, faSpinner, faCoins } from '@fortawesome/free-solid-svg-icons';
import ReactDOM from 'react-dom';

library.add(faCoins, faShoppingCart, faSpinner);

Shop.prototype.componentDidMount = () => {};

Enzyme.configure({ adapter: new Adapter() });

const context = {
    userInfo: {
        username: 'Test User',
        money: '150.00'
    },
    currentShoppingItem: {
        title: 'Current Item',
        Img: 'No Img',
        price: '10.00',
        desription: 'A Description',
        index: 0,
        id: 1
    },
};

describe('UI renders correctly', () => {
    it('renders content without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Router>
            <ProductsProvider value={context}>
                <Shop
                    history={{location: ''}}/>
            </ProductsProvider>
            </Router> , div);
    });
      
    it('Shop renders correctly', () => {
        const tree = makeShopWrapper();
        expect(tree).toMatchSnapshot();
    });
})

function makeShopWrapper() {
    return mount(
            <Router>
                <ProductsProvider value={context}>
                    <Shop
                        history={{location: ''}}/>
                </ProductsProvider>
            </Router>)
}
