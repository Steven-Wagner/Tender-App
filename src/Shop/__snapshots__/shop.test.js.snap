// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI renders correctly Shop renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Shop
      history={
        Object {
          "location": "",
        }
      }
    >
      <div
        className="page-wrapper shop-wrapper"
      >
        <AnnoyingAd />
        <ShopInfoBar
          handleDone={[Function]}
          totalMoney="150.00"
        >
          <div
            className="shopping-page-info"
          >
            <div>
              <TotalMoney
                totalMoney="150.00"
              >
                <div
                  className="total-money"
                >
                  <p>
                    <FontAwesomeIcon
                      border={false}
                      className="coins"
                      fixedWidth={false}
                      flip={null}
                      icon="coins"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-coins fa-w-16 coins"
                        data-icon="coins"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                     
                    150.00
                  </p>
                </div>
              </TotalMoney>
            </div>
            <div>
              <button
                className="done-button"
                onClick={[Function]}
              >
                Done
              </button>
            </div>
          </div>
        </ShopInfoBar>
        <CurrentProduct
          handleBuy={[Function]}
          handleLike={[Function]}
          handleSkip={[Function]}
          item={
            Object {
              "Img": "No Img",
              "desription": "A Description",
              "id": 1,
              "index": 0,
              "price": "10.00",
              "title": "Current Item",
            }
          }
        >
          <section
            className="current-product"
          >
            <div
              className="choose-buttons"
            >
              <button
                onClick={[Function]}
              >
                Skip
              </button>
              <button
                className="buy-button"
                onClick={[Function]}
              >
                Buy
              </button>
            </div>
            <h2>
              Current Item
            </h2>
            <p
              className="price"
            >
              Price: 
              10.00
            </p>
            <p
              className="description"
            />
          </section>
        </CurrentProduct>
      </div>
    </Shop>
  </Router>
</BrowserRouter>
`;
